### DISCUSSION TITLE
Incorrect state The mirai dispatcher is not running. Please call the start[help]

### DISCUSSION THREAD

**Discussion by MarcellGranat** [ropensci/targets#1225] [Help] [answered]
### Help

- [X] I understand and agree to <https://books.ropensci.org/targets/help.html>.

### Description

Hi,

My issue is that I am constatnly running into the following error:
> Error: ! Last error: 'errorValue' int 11 | Incorrect state The mirai dispatcher is not running. Please call the start() method of the controller (e.g. your_controller$start() before using methods like push(), collect(), and scale(). If you already did, then the dispatcher may have exited too early.

I cannot create a reprex for it because the project is already very large and contains sensitive data, but the triggering problem is that a pattern involves a large number of small targets, and the objects are saved in an onedrive folder (I assume this is the problem).

Furthermore, the error only occurs on Windows and not on Mac, and only in the case of the many small patterns.

**Comment by wlandau** [comment, marked as answer]
The advice at https://github.com/ropensci/targets/discussions/1217#discussioncomment-8284166 could help. Other than that, you could try upgrading `nanonext` and `mirai` to their CRAN versions and `crew` and `targets` to their current development versions:

```r
remotes::install_github("wlandau/crew")
remotes::install_github("ropensci/targets")
```

> triggering problem is that a pattern involves a large number of small targets

A reprex in your case might look like this. Does it reproduce the issue on your end?

```r
library(targets)
controller <- crew_controller_local(workers = 4)
tar_option_set(controller = controller)
list(
  tar_target(x, seq_len(1000)),
  tar_target(y, Sys.sleep(0.1), pattern = map(x)) # Adjust the seconds in Sys.sleep() as needed.
)
```

> the objects are saved in an onedrive folder

This is very likely to cause problems in general. Best to put the project (or at least the data store) in fast storage without a lot of temporary files flying around.
